### PR TITLE
Add shadow-dom

### PR DESCRIPTION
**Pull request by toddkao**
Add shadow dom

shadow dom

### CODE REVIEW

File: examples/nextjs/styles/globals.css [hunk 1]
```diff
@@ -7,3 +7,7 @@ body {
   -moz-osx-font-smoothing: grayscale;
   background-color: gray;
 }
+
+* {
+  font-size: 100px;
+}
```
Comment (toddkao) [inline examples/nextjs/styles/globals.css:13]: Testing to make sure css reset will override this

**Comment by codingki** [comment]
the deployment preview not picking up the fonts

**Comment by toddkao** [comment]
> the deployment preview not picking up the fonts

yeah... looks like the current way the custom fonts are imported does not work with shadow DOM either

File: packages/widget/package.json [hunk 2]
```diff
@@ -89,6 +89,7 @@
     "@radix-ui/react-scroll-area": "^1.0.5",
     "@radix-ui/react-switch": "^1.0.3",
     "@radix-ui/react-tooltip": "^1.0.7",
+    "@rollup/plugin-url": "^8.0.2",
```
Comment (codingki) [inline packages/widget/package.json:92]: this should be in dev dependency

File: packages/widget/rollup.config.js [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
 
 export default [
   {
     input: ['./src/index.ts'],
-    external: ['react', 'react-dom', '@r2wc/react-to-web-component'],
+    external: [
+      'react',
+      'react-dom',
+      '@r2wc/react-to-web-component',
+      'react-shadow-scope',
```
Comment (codingki) [inline packages/widget/rollup.config.js:15]: I think `react-shadow-scope` no need to be in external

**Review by codingki — APPROVED** [review]
rebase, and good to go

**Pull request merged**